Make aten.copy preserve strides (hf_Longformer)

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89464

Fixes https://github.com/pytorch/pytorch/issues/93428

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

Differential Revision: [D41460986](https://our.internmc.facebook.com/intern/diff/D41460986)

cc @mlazos @soumith @voznesenskym @yanboliang @penguinwu @anijain2305 @EikanWang @jgong5 @Guobing-Chen @chunyuan-w @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @desertfire

ezyang committed:
Make aten.copy preserve strides (hf_Longformer)

Fixes https://github.com/pytorch/torchdynamo/issues/1888

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

ezyang committed:
Update on "Make aten.copy preserve strides (hf_Longformer)"


Fixes https://github.com/pytorch/torchdynamo/issues/1888

Signed-off-by: Edward Z. Yang <ezyangfb.com>

[ghstack-poisoned]

bdhirsh reviewed (approved):
There's a chance this might mobile, if they have the `copy_` -> `view_copy` decomp baked in anywhere.

ezyang commented:
thx, I'll import

ezyang commented:
@pytorchbot merge

ezyang commented:
@ezyang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41460986).

ezyang committed:
Update on "Make aten.copy preserve strides (hf_Longformer)"


Fixes https://github.com/pytorch/torchdynamo/issues/1888

Signed-off-by: Edward Z. Yang <ezyangfb.com>

Differential Revision: [D41460986](https://our.internmc.facebook.com/intern/diff/D41460986)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41460986).

ngimel commented on aten/src/ATen/native/Copy.cpp:
```diff
@@ -278,27 +278,39 @@ static Tensor & copy_impl(Tensor & self, const Tensor & src, bool non_blocking)
+  TORCH_INTERNAL_ASSERT(nbytes % dtype_size == 0);
+  auto numel = nbytes / dtype_size;
+  auto self_full_size = self.as_strided_symint({numel}, {1}, 0);
+  auto clone = self_full_size.clone();
```
are we ok with this potentially using much more memory than it should?

ezyang replied:
It won't matter in inductor because inductor will decompose throwing out the storage size.

ezyang commented:
@ezyang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41460986).

ezyang committed:
Update on "Make aten.copy preserve strides (hf_Longformer)"


Fixes https://github.com/pytorch/torchdynamo/issues/1888

Signed-off-by: Edward Z. Yang <ezyangfb.com>

Differential Revision: [D41460986](https://our.internmc.facebook.com/intern/diff/D41460986)

[ghstack-poisoned]

ezyang committed:
Update on "Make aten.copy preserve strides (hf_Longformer)"


Fixes https://github.com/pytorch/torchdynamo/issues/1888

Signed-off-by: Edward Z. Yang <ezyangfb.com>

Differential Revision: [D41460986](https://our.internmc.facebook.com/intern/diff/D41460986)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Meta employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D41460986).

ezyang commented:
@pytorchbot merge -f "I had a green CI, just squashed earlier patch into this one"
